rmw_zenoh_cpp/src/detail/liveliness_utils.hpp

```diff
@@ -270,6 +269,29 @@ struct equal_to<rmw_zenoh_cpp::liveliness::ConstEntityPtr>
     return lhs->keyexpr_hash() == rhs->keyexpr_hash();
   }
 };
+
+template<>
+struct hash<rmw_zenoh_cpp::Gid>
+{
+  std::size_t operator()(const rmw_zenoh_cpp::Gid & gid) const noexcept
+  {
+    // This function implemented FNV-1a 64-bit as the GID is small enough
+    // (e.g. as of commit db4d05e of ros2/rmw RMW_GID_STORAGE_SIZE is set to 16)
+    // and FNV is known to be efficient in such cases.
+    //
+    // See https://github.com/ros2/rmw/blob/db4d05e/rmw/include/rmw/types.h#L44
+    // and https://datatracker.ietf.org/doc/html/draft-eastlake-fnv-17.html
+    static constexpr uint64_t FNV_OFFSET_BASIS_64 = 0x00000100000001b3;
+    static constexpr uint64_t FNV_PRIME_64 = 0xcbf29ce484222325;
```

Comment:
So we are already using xxhash3 in [liveliness_utils](https://github.com/ros2/rmw_zenoh/blob/bda2d7181d7e1ea2f7d143fabf7c4ffc6c670b20/rmw_zenoh_cpp/src/detail/liveliness_utils.cpp#L438-L439) to take a string and hash it into a 128-bit quantity.  Do we need to hash again?  In other words, could we just use the 128-bit GID directly and use that as the key?

Reply:
I believe [`std::hash`](https://en.cppreference.com/w/cpp/utility/hash) requires a `size_t`.

Reply:
We talked about this today and one optimization to make is to rely on FNA-1a to store the hash of the liveliness string as a `std::size_t` [here](https://github.com/ros2/rmw_zenoh/blob/bda2d7181d7e1ea2f7d143fabf7c4ffc6c670b20/rmw_zenoh_cpp/src/detail/liveliness_utils.cpp#L445) and send this size_t over any attachments as opposed to the 16B `Gid`.

Reply:
After taking a closer look at this, things are more complicated than they seem. FNV only  outperforms XXH3 for very small keys (see https://github.com/Cyan4973/xxHash/wiki/Performance-comparison#benchmarks-concentrating-on-small-data-), so we would be taking a hit. So it may be better to simply move to a 64-bit version of XXH3, as long as we agree that GID can be 8 bytes instead of 16.